### PR TITLE
dwr-512: adding compatible linux,spidev in dts

### PR DESCRIPTION
**Pull request by fid0did0**
The Dlink-dwr-512 mounts the SLIC device Si3210 controlled by the
second spi channel. Currently the generic spidev driver is attached.
To enable the spidev (using the linux,spidev compatible) a dedicated
patch is required.
The patch is also useful for the following .dts:
WRTNODE2.dtsi, VOCORE.dtsi, LINKIT7688.dts, MZK-DP150N.dts,
WIDORA-NEO.dts, PBR-D1.dts.
Without the patch the following error is found in dmesg:
	WARNING: CPU: 0 PID: 413 at drivers/spi/spidev.c:719 0x80cd433c()

Signed-off-by: Giuseppe Lippolis <giu.lippolis@gmail.com>

### CODE REVIEW

**Comment by thagabe** [comment]
Thank You! This was the spidev1 I was trying to add for the VoCORE dts. @blogic merge this!

**Comment by blogic** [comment]
no, it circumvents a warning that has been put in place specifically. please figure out how to runtime attach the spidev tot he spi.1 bus.

**Comment by fid0did0** [comment]
Dear blogic,
I cannot understand the warning rational.
I have a device with an hardwired device controlled by the spi1 and in my opinion make sense to export the spi control to the user space.
In addition I need to tell you that in the spidev.c are registered others two "compatible" string:
+ 	{ .compatible = "rohm,dh2228fv" },
+ 	{ .compatible = "lineartechnology,ltc2488" },

Why for these two devices the warning doesn't apply?
Can you please clarify these points?

**Comment by blogic** [comment]
you will need to go and ask the kernel folks that put this in place to explain. i recall there being a thread but i dont recall the actual reasoning. there is a clean and expected way to do this, please try to figure out how it works

i am guessing you will have to write some magic to a file under /sys

**Comment by fid0did0** [comment]
Dear blogic,
I ask to the spi linux subsystem the rationale of the warning.
They told me that the strategy is to keep the HW configuration described by the DT correct, also if there is no specific driver for the device.
Therefore they advice me to add a line using the specific device in the spidev.c.
(see the discussion here: http://marc.info/?t=148145791900001&r=1&w=2)
If in the future a specific driver will be available this line shall be removed and the DT will be unmodified.

So now I alligned the patch with this strategy.

My plan is currently to play in the userspace to control the device (providing some simple application) and over very long term try to design a specific driver.

Because the device I'm addressing is a SLIC telephone interface found in several SOHO router I will propose this patch currently only in lede and not in the upstream.

I'm looking forward to hearing from you.
Bye

File: target/linux/ramips/dts/DWR-512-B.dts [hunk 1]
```diff
@@ -104,6 +104,19 @@
 	};
 };
 
+&spi1 {
+	status = "okay";
+
+        spidev@1 {
```
Comment (mkresin) [inline target/linux/ramips/dts/DWR-512-B.dts]: Have have spaces in that node which should be tabs.

**Comment by mkresin** [comment]
Please squash you commits into one and drop that merge commit.

**Comment by fid0did0** [comment]
Hi Mathias,

how can I do it?

Shall I open a new PR, or is there a simpler way?

 

Bye.

 

Von: Mathias Kresin [mailto:notifications@github.com] 
Gesendet: Dienstag, 13. Dezember 2016 23:05
An: lede-project/source <source@noreply.github.com>
Cc: Giuseppe Lippolis <giu.lippolis@gmail.com>; Author <author@noreply.github.com>
Betreff: Re: [lede-project/source] dwr-512: adding compatible linux,spidev in dts (#604)

 

Please squash you commits into one and drop that merge commit.

**Comment by mkresin** [comment]
Opening a new PR is the easiest way. But it's annoying since we lose all comments and the reviews.

Please follow https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History to squash your commits and force update the branch of your github repository afterwards. The PR gets updated automatically:

```
git push origin +name-of-github-branch
```

**Comment by fid0did0** [comment]
Hi Mathias,
following your link I have now in my local repository the single squashed commit.
* 348bd39 (HEAD -> dwr-spi) spidev: adding registration for si3210
* 011f2c2 brcm2708: update linux 4.4 patches to latest version

Now I'm confused about how to procced.
In the link is clearly stated:
"Don’t include any commit you’ve already pushed to a central server – doing so will confuse other developers by providing an alternate version of the same change."

What shall I do now?

**Comment by blogic** [comment]
the branch is still broken, holding 4 and not 2 patches. if you don't know how to rebase/fold the branch, then close the PR and and send a new/clean one please

**Comment by blogic** [comment]
i have rebased this patch for you and added it to my staging tree

**Comment by fid0did0** [comment]
Hi blogic.
Can I ask you when you will commit this PR?

Bye.

**Comment by blogic** [comment]
https://git.lede-project.org/?p=source.git;a=search;s=Giuseppe+Lippolis;st=author

**Comment by fid0did0** [comment]
Ok you renamed the patch 
target/linux/generic/patches-4.4/850-add-linux-spidev-compatible.patch 
from 850 to 105.

It is ok from my side, but can I ask you why?
Just for my lerning.

Bye.

**Comment by bobafetthotmail** [comment]
patch numbers have a meaning explained in a text file somewhere in the source and also here https://lede-project.org/docs/guide-developer/adding_new_device#patches

**Comment by blogic** [comment]
i planned to move it to ralink/ but apparently forgot to push that patch. anyhow i just started to untangle all the patches and will fix this will at the job

**Comment by fid0did0** [comment]
Alberto thanks for the link.
Just for info:
currently my plan is to keep the patch only in lede and not push it in the upstream.
This is justified because the si3210 is a SLIC device tipically found in "router" devices.
Therefore until a "final" specific driver for the si3210 is not available, in my opinion doesn't make sense to update the upstream with a "generic" driver.

Therefore I named the patch 850: 8xx - other drivers.

Bye.

**Pull request closed without merging**